pkg/services/ngalert/remote/compat.go

```diff
@@ -0,0 +1,13 @@
+package remote
+
+import (
+	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/remote/client"
+)
+
+func PostableUserConfigToGrafanaAlertmanagerConfig(config *definitions.PostableUserConfig) *client.GrafanaAlertmanagerConfig {
+	return &client.GrafanaAlertmanagerConfig{
+		TemplateFiles:      config.TemplateFiles,
+		AlertmanagerConfig: config.AlertmanagerConfig,
+	}
+}
```

Comment:
We can avoid heap escapes by not returning a pointer here. Consider that this will run inside a loop.

```suggestion
func PostableUserConfigToGrafanaAlertmanagerConfig(config *definitions.PostableUserConfig) client.GrafanaAlertmanagerConfig {
	return client.GrafanaAlertmanagerConfig{
		TemplateFiles:      config.TemplateFiles,
		AlertmanagerConfig: config.AlertmanagerConfig,
	}
}
```

I tried this change locally and got rid of all heap escapes for `GrafanaAlertmanagerConfig` after some minor modifications to the tests.